docs-starlight/src/content/docs/05-troubleshooting/03-performance.mdx

```diff
@@ -0,0 +1,149 @@
+
+However, given the nature of the problems that Terragrunt solves, there are some obvious wins that you can make without measuring performance, if you're aware of the tradeoffs.
+
+### Provider Cache
+
+One of the most expensive things that OpenTofu/Terraform does, from a bandwidth and disk utilization perspective, is download and install providers. These are large binary files that are downloaded from the internet, and not cached across units by default.
+
+You can significantly reduce the amount of time taken by Terragrunt runs by enabling the provider cache server, like this:
+
+```shell
+terragrunt run --all plan --provider-cache
+```
+
+#### Provider Cache - Gotchas
+
+The provider cache server is a single server that is used by all Terragrunt runs being performed in a given Terragrunt invocation. You will see the most benefit if you are using it in a command that will perform multiple OpenTofu/Terraform operations, like with the `--all` flag and the `--graph` flag.
+
+When performing individual runs, like `terragrunt plan`, the provider cache server can be a net negative to performance, because starting and stopping the server might add more overhead than just downloading the providers. Whether this is the case depends on many factors, including network speed, the number of providers being downloaded, and whether or not the providers are already cached in the Terragrunt provider cache.
+
+When in doubt, [measure the performance](#measuring-performance) before and after enabling the provider cache server to see if it's a net win for your use case.
+
+We are coordinating with the OpenTofu team to improve the behavior of concurrent access to provider cache, so that this flag will be unnecessary for most users in the future.
+
+See [#1939](https://github.com/opentofu/opentofu/issues/1483) for more details.
+
+### Fetching Output From State
+
+Under the hood, Terragrunt dependency blocks leverage the OpenTofu/Terraform `output -json` command to fetch outputs from one unit and leverage them in another.
+
+The OpenTofu/Terraform `output -json` command does a bit more work than simply fetching output values from state, and a significant portion of that slowdown is loading providers, which it doesn't really need in most cases.
+
+You can significantly improve the performance of dependency blocks by using the `--dependency-fetch-output-from-state` flag. When the flag is set, Terragrunt will directly fetch the backend state file from S3 and parse it directly, avoiding any overhead incurred by calling the `output -json` command.
+
+For example:
+
+```shell
+terragrunt run --all plan --dependency-fetch-output-from-state
+```
+
+#### Fetching Output From State - Gotchas
+
+The first thing you need to be aware of when considering usage of the `--dependency-fetch-output-from-state` flag is that it only works for S3 backends. If you are using a different backend, you should not use this flag.
```

Comment:
s/should not/can not/?  What happens if they do, i assume it blows up?

Reply:
Na. It just ignores the flag.